linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -4389,6 +4415,14 @@ private[optimizer] abstract class OptimizerCore(
               PreTransLit(IntLiteral(_))) if (y & 31) != 0 =>
             foldBinaryOp(Int_>>>, lhs, rhs)
 
+          case (PreTransBinaryOp(op @ (Int_| | Int_& | Int_^),
+              PreTransLit(IntLiteral(x)), y),
+              z @ PreTransLit(IntLiteral(zValue))) =>
+            foldBinaryOp(
+                op,
+                PreTransLit(IntLiteral(x >> zValue)),
+                foldBinaryOp(Int_>>, y, z))
```

Comment:
Just to check my understanding: This transformation is useful because if `x >> zValue` "shifts-out" all the bits, we can subsequently constant fold `op` away?

Reply:
Yes. We were already doing the same for `Int_>>>`, which was beneficial for the standard `RuntimeLong.*` when `a` is constant and `0 <= a <= 0xffff`. This is the equivalent optimization that is beneficial for `multiplyFull`.